### ISSUE TITLE
(Dashboard) - Errors when deleting items from app

### ISSUE THREAD

**Issue by caseychoiniere** [Duke-Translational-Bioinformatics/duke-data-service-portal#833]
Errors are thrown in mainStore.jsx when deleting items. This looks like it's happening because you're 
calling `dashboardStore.removeDownloadedItem(id, parentId)` at line 515 in mainStore.jsx. This works fine when you;re in the dashboard but deleting items anywhere else in the app fails now because `childrenIds` are not set in  dashboardStore. I'm assuming this affects all other areas of the app as well such as moving items, editing etc. 

You'll need to make some kind of check to keep this from firing.

Also want to click test the app anywhere you've injected this logic between the dashboard and the rest of the app.

I'm actually surprised the test didn't fail here? Something else to look at.

```javascript
 @action deleteItemSuccess(id, parentId, path) {
        this.loading = false;
        dashboardStore.removeDownloadedItem(id, parentId)
        this.listItems = this.listItems.filter(l => l.id !== id);
        this.totalItems--;
        if(this.listItems.length === 0) this.getChildren(parentId, path)
    }
````

error:
```
ex = TypeError: Cannot read property 'childrenIds' of undefined at DashboardStore.removeDownloadedItem (webpack:///./app/scripts/stores/dashboardStore.js?:138:33) at executeAction (webpack:///./~/mobx/lib/mobx.js?:915:19) at DashboardStore.res (webpack:///./~/mobx/lib/mobx.js?:906:16) at MainStore.deleteItemSuccess (webpack:///./app/scripts/stores/mainStore.js?:783:38) at executeAction (webpack:///./~/mobx/lib/mobx.js?:915:19) at MainStore.res (webpack:///./~/mobx/lib/mobx.js?:906:16) at eval (webpack:///./app/scripts/stores/mainStore.js?:759:25) at <anonymous>
```

**Comment by ChetCorey** [comment]
Yea, I will add tests to all of my code, and write them in going forward.
Thanks for catching all this junk

**Comment by ChetCorey** [comment]
This is resolved in commit #730a2aec75335ffab6280a14d7aeba6e44523aed

![success dashboard - errors when deleting items from app 833](https://user-images.githubusercontent.com/7772027/39017036-2b4a76c8-43f0-11e8-89e9-cb352f0af549.gif)
